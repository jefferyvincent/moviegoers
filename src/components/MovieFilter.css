.filter {
    float: left;
    width: 300px;
    margin: 0 1% 0 0;
    position: relative;
    transition: .5s;
}

.filter-selected {
    border: 1px solid #eee;
    padding: 4% 4% 4% 12%;
    border-radius: 2px;
    color: #666;
    font-size: 16px;
    font-weight: 100;
    cursor: pointer;
    transition: .5s;
}

.filter-selected img {
    margin-left: 1rem;
}

.filter-options {
    position: absolute;
    top: 46px;
    left: 0;
    background: #fff;
    width: 100%;
    z-index: 1;
    margin: 0;
    padding: 4% 0;
    border-right: 1px solid #eee;
    border-left: 1px solid #eee;
    border-bottom: 1px solid #eee;
    max-height: 300px;
    overflow: auto;
}

.filter-options li {
    display: block;
    padding: 4% 12%;
    transition: .5s;
    cursor: pointer;
    color: #666;
    font-weight: 100;
    border-right: 1px solid #fff;
    border-left: 1px solid #fff;
}

.filter-options li:hover {
    color: #00c8ff;
    background-color: black;
    text-decoration: none;
}

.filter-text {
    font-size: 1rem;
    color: grey;
}

.movieFilter {
    width: 100%;
}

@media only screen and (max-width: 904px) {
    .filter{
        width: 100%;
    }
}