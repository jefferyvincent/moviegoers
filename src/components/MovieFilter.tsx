import React, {useState, useContext} from 'react';
import Context from '../Context';

import arrow from '../img/arrow.svg';
import './MovieFilter.css';

interface MovieFilterProps {
    filters: Array<object>;
    active_filter: string;
    history: any;
}

const MovieFilter: React.FC<MovieFilterProps> = ({ filters, active_filter, history}) => {
    const dispatch:any = useContext(Context);
    const [active, setActive] = useState('Now Playing');
    const [hidden, setHidden] = useState('hidden');
    
    const filterResults = (filter:string):void => {
        dispatch({
            type: 'updateFilter',
            payload: filter
        });
        dispatch({
            type: 'updatePage',
            payload: 1 
        });
        toggleFilterOptions();
        history.push('/movies/1');
    }

    const toggleFilterOptions = ():void => {
        if (hidden === "hidden") {
            setHidden('');
            document.addEventListener('mousedown', hideFilter, false);
        } else {
            setHidden('hidden');
        }
    }

    const hideFilter = ():void => {
        setTimeout(()=>{
            setHidden('hidden');
            document.removeEventListener('mousedown', hideFilter, false);
        }, 500)
    }

    return (
        <div className="movieFilter">
            <h3 className="filter-text">Showing Movies sorted by {active}</h3>
            <div className="filter">
                <div className="filter-selected" onClick={toggleFilterOptions}>
                    {active} <img src={arrow} alt="filter" />
                </div>
                <ul className={"filter-options " + hidden}>
                    {
                        filters.map((item:any, key:number)=> {
                            if (active_filter !== item.name) {
                                return (
                                    <li key={key} onClick={()=> {filterResults(item.name); setActive(item.title)}}> {item.title}</li>
                                )
                            } else {
                                return null;
                            }
                        })
                    }
                </ul>
            </div>
        </div>
    );
}

export default MovieFilter;