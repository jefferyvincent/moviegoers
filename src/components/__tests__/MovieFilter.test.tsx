import React from 'react';
import { shallow } from 'enzyme';
import MovieFilter from '../MovieFilter';

describe('<MovieFilter/>', () => {
  let wrapper:any;
  const filter_data = [
    {
      name: 'now_playing',
      title: 'Now Playing'
    },
    {
      name: 'popular',
      title: 'Popular'
    },
    {
      name: 'top_rated',
      title: 'Top Rated'
    }
  ];


  beforeEach(() => { wrapper = shallow(<MovieFilter history={[]} filters={filter_data} active_filter="now_playing"/>); });
  it('renders a filter', () => {
    expect(wrapper.find('div.movieFilter').length).toEqual(1);
  });

  it('should show the hidden ul filter list when clicked', () => {
    expect(wrapper.find('ul').hasClass('hidden')).toEqual(true);
    wrapper.find('div.filter-selected').simulate('click');
    expect(wrapper.find('ul').hasClass('hidden')).toEqual(false);
    });
});