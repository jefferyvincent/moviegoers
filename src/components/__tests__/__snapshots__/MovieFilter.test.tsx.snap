// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieFilter/> renders 1`] = `
<DocumentFragment>
  <div
    class="movieFilter"
  >
    <h3
      class="filter-text"
    >
      Showing Movies sorted by Now Playing
    </h3>
    <div
      class="filter"
    >
      <div
        class="filter-selected"
      >
        Now Playing 
        <img
          alt="filter"
          src="arrow.svg"
        />
      </div>
      <ul
        class="filter-options hidden"
      >
        <li>
           Popular
        </li>
        <li>
           Top Rated
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
